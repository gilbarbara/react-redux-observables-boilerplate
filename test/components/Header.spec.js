import React from 'react';
import { mount } from 'enzyme';

import Header from 'components/Header';

jest.mock('react-router-dom/NavLink', () =>
  (({ to, children }) => (<a href={to}>{children}</a>))
);

const mockDispatch = jest.fn();

function setup() {
  const props = {
    dispatch: mockDispatch,
    user: {
      isAuthenticated: false,
    },
  };

  return mount(<Header {...props} />);
}

describe('Header', () => {
  const wrapper = setup();

  it('should be a Component', () => {
    expect(wrapper.instance() instanceof React.Component).toBe(true);
  });

  it('should render properly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should handle login click', () => {
    wrapper.find('.app__header__login').simulate('click');
    expect(mockDispatch.mock.calls[0][0]).toEqual({
      type: 'USER_LOGIN_REQUEST',
      payload: {},
    });
  });

  it('should handle logout click', () => {
    wrapper.setProps({
      user: {
        isAuthenticated: true,
      },
    });

    wrapper.find('.app__header__logout').simulate('click');
    expect(mockDispatch.mock.calls[1][0]).toEqual({
      type: 'USER_LOGOUT_REQUEST',
      payload: {},
    });
  });
});

