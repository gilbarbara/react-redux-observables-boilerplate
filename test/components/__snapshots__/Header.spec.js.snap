// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render properly 1`] = `"<header class=\\"app__header\\"><div class=\\"app__container\\"><ul class=\\"app__header__menu\\"><li><a href=\\"/\\" class=\\"app__header__link\\">Home</a></li><li><a href=\\"/private\\" class=\\"app__header__link\\">Private</a></li><li><button class=\\"app__header__login btn btn-sm btn-primary btn-icon\\"><i class=\\"i-sign-in\\"></i><span>Login</span></button></li></ul></div></header>"`;
